pennylane_lightning/lightning_qubit/lightning_qubit2.py

```diff
@@ -21,135 +21,184 @@
 
 import pennylane as qml
 from pennylane.devices import Device, ExecutionConfig, DefaultExecutionConfig
-from pennylane.devices.preprocess import decompose, validate_device_wires, decompose, validate_measurements, validate_observables, no_sampling
+from pennylane.devices.preprocess import (
+    decompose,
+    validate_device_wires,
+    decompose,
+    validate_measurements,
+    validate_observables,
+    no_sampling,
+)
 from pennylane.devices.qubit.sampling import get_num_shots_and_executions
 from pennylane.tape import QuantumScript
 from pennylane.transforms.core import TransformProgram
 from pennylane.typing import Result, ResultBatch
 
+from ._state_vector import LightningStateVector
+from ._measurements import LightningMeasurements
 
-def dummy_simulate(circuit, rng=None, c_dtype=np.complex128, batch_obs=False, mcmc=False,
-    kernel_name="Local", num_burnin=100):
-    return tuple(0.0 for _ in circuit.measurements)
 
-def dummy_jacobian(circuit):
+try:
+    # pylint: disable=import-error, no-name-in-module
+    from pennylane_lightning.lightning_qubit_ops import (
+        StateVectorC64,
+        StateVectorC128,
+    )
+
+    LQ_CPP_BINARY_AVAILABLE = True
+except ImportError:
+    LQ_CPP_BINARY_AVAILABLE = False
+
+
+def simulate(circuit: QuantumScript, dtype=np.complex128, debugger=None) -> Result:
+    """Simulate a single quantum script.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        dtype: Datatypes for state-vector representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        debugger (_Debugger): The debugger to use
```

Comment:
I think we should probably leave `_Debugger` and snapshots out of scope for the project for now.  This could probably be added in later as part of a separate project and feature.

Reply:
I think this makes sense, my idea here was to have it default to None, just to know where to act in the future.
Do you prefer if I remove it? Could you please clarify what you mean by snapshots in this context? Am I referring to it anywhere?

Reply:
I will remove the debugger references.